crates/core/tedge_agent/src/software_manager/actor.rs

```diff
@@ -229,6 +243,48 @@ impl SoftwareManagerActor {
         self.output_sender.send(response.into()).await?;
 
         self.state_repository.clear().await?;
+        Self::detect_self_update()?;
```

Comment:
I would not do that inside the `handle_software_update_operation()` but at the call site.

Reply:
Added this check specifically in this handler to limit it to software **update** requests only and **not** on software **list** requests.

Reply:
> Added this check specifically in this handler to limit it to software **update** requests only and **not** on software **list** requests.

Yeah, I wondered the same. But, I think that having this check is good at the top level two reasons:
- First, this is simpler
- Second, this is even better as this will be helpful if `tedge-agent` is directly updated by-passing the software update feature.